File: 01_Recursion/03_tower_of_hanoi.py
def tower_of_hanoi( n, frompeg,  topeg,  auxpeg):
    if (n==1):
        print(f"Move disk 1 from peg {frompeg} to peg {topeg}")
        return

    tower_of_hanoi(n-1,frompeg, auxpeg, topeg)

    print(f"\n move disk {n} from peg {frompeg} to peg {topeg} \n")

    tower_of_hanoi(n-1, auxpeg, topeg, frompeg)

n=int(input("Enter number of disks: "))


tower_of_hanoi(n, 'A', 'C', 'B')

    

File: 01_Recursion/03_tower_of_hanoi.c
#include <stdio.h>
void tower_of_hanoi(int n, char frompeg, char topeg, char auxpeg){
if (n==1){
    printf("Move disk 1 from peg %c to peg %c", frompeg, topeg);
    return;
}

tower_of_hanoi(n-1,frompeg, auxpeg, topeg);

printf("\n move disk %d from peg %c to peg %c \n", n, frompeg, topeg);

tower_of_hanoi(n-1, auxpeg, topeg, frompeg);
}


int main() {
    int n;

    printf("Enter number of disks: ");
    scanf("%d", &n);

    tower_of_hanoi(n, 'A', 'C', 'B');

    return 0;
}
